firmware/command.cpp

```diff
@@ -1,156 +1,548 @@
+#include "const.h"
 #include "command.h"
 #include "strutil.h"
-#include "const.h"
 #include "globals.h"
 #include "motion.h"
-
+#include "pins.h"
+int counter = 0;
 Command::Command( char *buf ) {
-	char * buf_ptr = buf;
-	this->cmd_class = '?';
-	this->cmd_id = 0;
-	this->x = position[0] * 100 / STEPS_PER_MM;
-	this->y = position[1] * 100 / STEPS_PER_MM;
-	this->e = 0;
-	this->f = modal_feedrate;
-	this->i = 0;
-	this->j = 0;
-	this->next = NULL;
-
-	#ifdef DEBUG_OUTPUT
-		Serial.print( F( "// Start position: (" ) );
-		Serial.print( position[0] / STEPS_PER_MM );
-		Serial.print( "," );
-		Serial.print( position[1] / STEPS_PER_MM );
-		Serial.println( ")" );
-	#endif
-	switch( *buf_ptr++ ) {
-		case 'F':
-			this->cmd_class = 'F';
-			this->f = get_float( &buf_ptr );
-			break;
-		case 'G':
-			parse_g_command( this, &buf_ptr, buf );
-			break;
-		default:
-			Serial.print( F("// warning: unknown command " ) );
-			Serial.println( buf );
-			Serial.flush();
-			break;
-	}
-	#ifdef DEBUG_OUTPUT
-		Serial.print( F( "// Instantiated command from buffer: " ) );
-		print_command( Serial, this );
-	#endif
+  char * buf_ptr = buf;
+  this->cmd_class = '?';
+  this->cmd_id = 0;
+
+  this->x = 0;
```

Comment:
Positions should be modal; that is, if I don't specify an X coordinate in a command, the default should be the current/previous X coordinate, rather than 0.